feat: add some more keyboard mappings

ByteBaker opened the pull request:
Part of #69 to support more key mappings

Also 
- bump `env_logger: 0.10 -> 0.11`, 
- add macros to generate code in `keyboard.rs` to conveniently generate mappings

pewsheen commented:
It looks good overall
I am just a little concerned about whether there are keys that are not 1:1 matched, then we'll need this old way back in the future...

ByteBaker commented:
Fair point. I think I'll just modify the macro to accept raw pattern blocks as well. This way if we ever need to, we can just add those patterns inside the macro invocation alongside the rest of the items as they're now.

ByteBaker commented:
I've modified the macros and added documentation.

Now they also support calls like `logical_to_winit_key!(key, Escape, F13 => F1)` to map custom variants when one-one mappings aren't available.

This one in example maps `Escape => Escape` and `F13 => F1`.

pewsheen reviewed (approved):
awesome
